core/types/subtyping.cc

```diff
@@ -1211,25 +1220,47 @@ bool isSubTypeUnderConstraintSingle(const GlobalState &gs, TypeConstraint &const
 
```

Comment:
I can't comment on the relevant line, but we're likely going to want to explain when !`classSymbolIsAsGoodAs` too. Since that's on classes not on types, it's likely good to use "derives from" instead of "is subtype of" in the error phrasing:

```
`{}` does not derive from `{}`"
```

This can also be a follow up in a future PR.